### DISCUSSION TITLE
[New Product Request]: 12 Step Event Participant Manager

### DISCUSSION THREAD

**Discussion by kiyote33** [code4recovery/12-step-meeting-list#1099] [Ideas]
### Contact Details

waynep@code4recovery.org

### Requested Feature/Enhancement

**Overview**
![image](https://user-images.githubusercontent.com/57125427/227009585-dca46b0d-fd32-4d09-a1ba-721f5f5503ba.png)
The 12 Step Event Participant Manager is envisioned as an Add-on plugin for the WordPress  “The Events Calendar” plugin, enhancing its capabiliy for a viewer to read about a specific Service Committee event and then choose to sign-up as an attendee and particpate in that event in one of the available roles (Lead Presenter, Co-lead, Participant, or Shadow).

The tool is intended to facilitate service committee planing and managing of events which require a certain level of experienced presenters who can deliver their organization’s recovery message in a comprehensive and consistent fashion. This allows for those new to the committee’s work to simply “Shadow”, watching and learning (i.e. “What we are”, “What we can do”, and “What we cannot do”). When they are ready, they can then choose to move on to grow in one of the more challenging roles of participating in a presentation and/or even leading the presentation.

The following screenshot images are taken from [this prototype application](https://edmontonaa.net/picpc-events). 

**Calendar & List Views**
![image](https://user-images.githubusercontent.com/57125427/227009930-35ee926c-18b9-4904-b99e-dba00d72151f.png)
![image](https://user-images.githubusercontent.com/57125427/227009989-1b6607e7-d547-40cc-a9c2-ac9e79d7cbf0.png)
![image](https://user-images.githubusercontent.com/57125427/227010082-e7717a91-4952-4791-9afe-fef365d5bf2f.png)

**View Event Detail**
Enhancements: 
1.	Link to new (admin only) View Booking Requests page
2.	Link to new Register as a Participant page
3.	Already signed up participants listing appended on this page
![image](https://user-images.githubusercontent.com/57125427/227010556-88fe56a1-94ad-4e71-82ba-7dc0e9a1bcae.png)

**View Booking Requests**
Enhancement Details
1.	Event Title
2.	Participant Request Listing (see below for details)
3.	Export Data →Excel
![image](https://user-images.githubusercontent.com/57125427/227010785-fc1d9300-2110-46aa-b6fd-35d31a4687dc.png)

**Create/Edit Event**
![image](https://user-images.githubusercontent.com/57125427/227011058-5460486d-cff8-448b-901a-335c0211aef0.png)

**Event Settings**
![image](https://user-images.githubusercontent.com/57125427/227011320-e8cf1b6a-1cb5-4b96-b239-af729ab631dd.png)
Note: The free version Event Calendar input screen has multi-day selection possible, but does not allow for repetition. Pro add-on plugin is required for repeating events (Pro is free for verified non-profit organizations).

**Event Details**
![image](https://user-images.githubusercontent.com/57125427/227011973-8ce997f4-f042-44a9-bfc2-cdf5273eff28.png)

**Event Participants**
![image](https://user-images.githubusercontent.com/57125427/227012118-2711c2fe-63e6-4ce4-b11f-08e2621db8a2.png)

**Booking Options**
![image](https://user-images.githubusercontent.com/57125427/227012307-b79e4fb4-6baa-48b4-8d67-70aaab1e106e.png)

**Comment by joshreisner** [comment]
neat idea! 🎉 

is there any action to be taken in 12 Step Meeting List? perhaps you might make this a discussion or start a new code for recovery repo for this project?

**Comment by kiyote33** [comment]
No, nothing in the proposal relates to the 12 Step Meeting List. I didn't think I was authorized to create a repo. As this is still a request, I wanted to  do a little project proposal documentation so others have a basis to provide feedback. I was tossing Discussion around in my head, but went with Issues as that's where I started with the Feedback Enhancement. I'm fine with moving it to Discussion as it doesn't have anything to do with TSML.

**Reply by joshreisner** [reply]
cool thanks! i checked your access and you should be good to create new repositories within the organization. good callout to discuss it as a team though - if you're there saturday would be great to hear your thoughts about it. also something perhaps to call out in the `#general` slack channel

**Comment by FredinMa** [comment]
H&I (aka Treatment) 12-Step Calls (meetings) are a specific use-case for this, IMO.  I'd be interested to see a WP plug-in to support the booking & management of Treatment Commitments (12-Step Calls) to hospitals & treatment facilities. I have done something with .NET but would love to see a WP plugin, which would ideally work for a [district](https://aadistrict26.org/D26Treatment_Page.html) , [IGCO](https://aaWorcester.org/Treatment_Page.html) and [Area](https://area30Treatment.org/A30TC_Treatment_Page.html) or all three.
